torchvision/prototype/models/vision_transformer.py

```diff
@@ -233,14 +235,40 @@ def forward(self, x: torch.Tensor):
         return x
 
 
+_COMMON_META = {
+    "categories": _IMAGENET_CATEGORIES,
+    "interpolation": InterpolationMode.BILINEAR,
+}
+
+
 class ViT_B_16_Weights(WeightsEnum):
-    # If a default model is added here the corresponding changes need to be done in vit_b_16
-    pass
+    ImageNet1K_V1 = Weights(
+        url="https://download.pytorch.org/models/vit_b_16-0413b9bf.pth",
+        transforms=partial(ImageNetEval, crop_size=224),
```

Comment:
Are you sure the weights are optimal without using resize_size optimizations?

Reply:
Currently doing a batch of new training to see whether the accuracy can be further increased by adopting the new recipe.

Reply:
Actually - i think the current results are publishable - they might not be the most optimal results, but they are good enough to beat the original paper's result and classy vision's benchmarking results. I admit that by adopting more fancy training techniques such as conv_stem or applying the tuned parameters in the new recipe can help us gain more points, but it does requires training more epochs (for the torchvision new training recipe it trains 600 epochs to converge). But for now, all the baseline results in other repo (such as google vit or deit) they always train 300 epochs (just as i did here), and i think we can keep aligned with them in imagenet1k_v1 (for reference). Probably in the future we can publish the new weights (trained with new recipe) and name it imagenet1k_v2. 
wdyt? @datumbox

Reply:
@sallysyw I think we can do better. There are multiple pros in converging towards a generalizable recipe on which we can build upon rather than having custom recipes per model, especially if those custom recipes achieve lower accuracies as it is in this case. 

TorchVision's new recipe (published [here](https://pytorch.org/blog/how-to-train-state-of-the-art-models-using-torchvision-latest-primitives/)), when combined with some of the ViT specific optimizations you introduced (such as gradient clipping and the RA sampler) and some proposals from the [community](https://github.com/pytorch/vision/pull/5084#issuecomment-998369197), achieves much better results than the ones included in this PR. 

Here are some quick runs I did to confirm that the new recipe works well on ViT:
```
vit_b_16 (jobid 11379, epoch 450)
Acc@1 80.730 Acc@5 95.228

vit_l_32 (jobid 12096, epoch 378)
Acc@1 75.050 Acc@5 91.824

vit_l_16 (jobid 12107/12349/12567, epoch 378)
Acc@1 79.668 Acc@5 94.660
```

Given the above have 0.7-0.8 point higher accuracy, we should prefer them. Of course you can further optimize them by tuning the hyper-parameters but we should at least deploy the best available models. I've included the jobids so that you can find them easier on the internal infra but happy to chat offline to help.

Reply:
Thanks for the info - I looked at the logs you provided and analyzed it using [this script](https://gist.github.com/kazhang/89525c80f2fe56caf25bb6779cf8cf33). And I found that the accuracy included in this PR are actually higher than the results suggested above - for example:

## 1. vit_b_16
### 1.1 The results trained with the new recipe (job-id 11379)
```
(base) sallysyw@ip-10-200-71-59:~/workspace$ python scripts/best_accuracy.py --job-ids 11379
Best acc@1      acc@5   epoch
78.764  93.334  588
Best EMA acc@1  acc@5   epoch
80.656  95.09   450
```

### 1.2 My previous result on vit_b_16 (job-id 10266)
```
(base) sallysyw@ip-10-200-71-59:~/workspace$ python scripts/best_accuracy.py --job-ids 10266
Best acc@1      acc@5   epoch
80.033  94.663  298
Best EMA acc@1  acc@5   epoch
81.082  95.32   268
```

## 2. vit_l_32
### 2.1 The results trained with the new recipe (job-id 12096)
```
(base) sallysyw@ip-10-200-71-59:~/workspace$ python scripts/best_accuracy.py --job-ids 12096
Best acc@1      acc@5   epoch
69.442  87.886  552
Best EMA acc@1  acc@5   epoch
75.05   91.824  378
```

### 1.2 My previous result on vit_b_32 (job-id 10265)
```
(base) sallysyw@ip-10-200-71-59:~/workspace$ python scripts/best_accuracy.py --job-ids 10265
Best acc@1      acc@5   epoch
75.646  92.164  296
Best EMA acc@1  acc@5   epoch
75.92   92.468  291
```

Looks like my best accuracy is always higher?
Is it because you are reporting the EMA accuracy but I'm reporting the non-EMA accuracy? If so, should I go with the EMA accuracy here?